# Hypothesis Test on Simple Regression

Maybe it is unexpected, but we **can** run a hypothesis test on a regression! This frequently is a test of a relationship between the predictor variable $x$ and the response variable $y$. 

&emsp; $H_0$: There is no relationship between $x$ and $y$. <br>
&emsp; $H_1$: There is some relationship between $x$ and $y$.

Mathematically, this is the same as:

&emsp; $H_0$: $\beta_1 = 0$ <br>
&emsp; $H_1$: $\beta_1 \neq 0$

Why? Because if the best model has $\beta_1 = 0$, then the best model is $Y = \bar{y}$ and *does not include $x$ at all*!

## Test Statisic

We previously learned about the t-statistic where we calculated it as:

$$ t = \frac{\text{difference from null hypothesis}}{\text{standard error}} $$

We are investigating the difference between our calculated best coefficient $\hat{\beta_1}$ and the unknown "true" best coefficient $\beta_1$, so we get $\hat{\beta_1}-0$ in the numerator. 

And as you likely guessed, we again use the sum of squares to determine our standard error by dividing our variance by the sum of squares of our predictors:

$$ \text{SE}(\hat{\beta_1}) = \sqrt{\frac{\sigma^2}{\sum_{i=1}^n (x_i - \bar{x})^2}} $$

So we get our t-statistic as:

$$ t = \frac{\hat{\beta_1}}{\text{SE}(\hat{\beta_1}) } $$

And now that we have a t test statistic, we can compute the area under a t-distribution to find the corresponding p-value and make a conclusion about our hypothesis test. Let's see how to do this in Python:

In [ ]:
import numpy as np
from sklearn.linear_model import LinearRegression
import statsmodels.api as sm

# Random data
X_random = np.random.rand(10,1)
y_random = X_random + np.random.rand(10,1)*(-1)**np.random.randint(0,2)

# Creating the linear regression is just two steps in Python!
reg_lin = LinearRegression()
reg_lin.fit(X_random, y_random)

X_random_2 = sm.add_constant(X_random) # Adds a constant coefficient column to estimate beta_0
estimate = sm.OLS(y_random, X_random_2)
estimate_2 = estimate.fit()

print(estimate_2.summary())

                            OLS Regression Results                            
Dep. Variable:                      y   R-squared:                       0.622
Model:                            OLS   Adj. R-squared:                  0.575
Method:                 Least Squares   F-statistic:                     13.17
Date:                Wed, 08 Jul 2026   Prob (F-statistic):            0.00669
Time:                        12:39:46   Log-Likelihood:                0.22884
No. Observations:                  10   AIC:                             3.542
Df Residuals:                       8   BIC:                             4.147
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.5636      0.158      3.561      0.0

There is a lot to unpack here! Let's ignore everything except this table:

|      | coef   | std err | t     | $P>\lvert t \rvert$ | $[0.025$   |   $0.975]$ |
| :--- | :---   | :---    | :---  | :---    | :--- | :--- |
|const | 0.5636 | 0.158   | 3.561 | 0.007   | 0.199   |    0.929 |
|x1    | 1.0009 | 0.276   | 3.629 | 0.007   | 0.365   |    1.637 |

Let's make the labels a little more descriptive:

|      | $\beta_i$ | standard error | t-statistic | p-value | left confidence interval value | right confidence interval value | 
| :--- | :---   | :---    | :---  | :---    | :--- | :--- |
|const | 0.5636 | 0.158   | 3.561 | 0.007   | 0.199   |    0.929 |
|x1    | 1.0009 | 0.276   | 3.629 | 0.007   | 0.365   |    1.637 |

The first column tells us the coefficients for our linear regression. In this case, we get 

$$ Y = 1.0009 x + 0.5636 $$

The second column tells us the standard error per variable, which we can then use to create the t-statistic (third column) and then integrate the t-distribution to find the p-value (fourth column). The last two columns give us the confidence interval of each coefficient. 

So for our hypothesis test purposes, we would report the p-value for x1. Since this is less than our significance level (choose your favorite significance level), we would **reject** the null hypothesis that there is no relation between our predictor variable $x$ and response variable $y$. We would then calculate effect size to determine how much of a relation there is. 

## Effect Size

In [ ]:
from statsmodels.stats.power import TTestPower

power_analysis = TTestPower()
effect_size = power_analysis.solve_power(
    effect_size = None, 
    nobs = len(X_random),
    alpha = 0.05,
    power = 0.9, 
    alternative = 'two-sided'
)

print(effect_size)

1.1545650520269477


This tells us there is a strong relationship between our predictor variable and response variable.